cat/opencl.cat

```diff
@@ -0,0 +1,147 @@
+OpenCL
```

Comment:
I think this cat file is a good case where it would be really helpful to add support for dartagnan to define things like this in the cat file
```
let name(r) = r op ...
```
This would greatly simply the cat model.

Implementation-wise, one can simply add some new classes `/wmm/definition/Parametric.java` which contains a deifntiion of the relation and `/wmm/definition/ParametricCall.java` which contains a `Parametric` object (specifying the definition of this call) and the relation it is applied to. The first one would be related to definitions of the form `let name(r) = ...` and the latter to the usage `... = name(co) ...`. 

Then you can have a processing pass `/wmm/processing/...` that would apply to the specific relation into the definition.

I know this one is based in the one from herd repo, but it would also make sense to add yet another cat file following the paper. There are some differences between both versions that are not trivial, and this would be useful to see if both models are really equivalent

Reply:
- Parametric implemented.
- There seems to be four versions: 2 from herd and 2 from the paper. The difference between herd and the paper is not clear for me. I will to try adding all of them as different cats and test the difference.